### ISSUE TITLE
Variable not defined in unreachable code with semantic highlighting on

### ISSUE THREAD

**Issue by karthiknadig** [microsoft/pylance-release#406]
## Environment data

-   Language Server version: 2020.9.5
-   OS and version: Windows
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9

## Expected behaviour

Should not be showing squiggles on `expected`.

## Actual behaviour

![image](https://user-images.githubusercontent.com/3840081/94068193-b8ca8c80-fda3-11ea-9a1a-03bdab67a4c0.png)

## Logs

```
XXX
```

## Code Snippet / Additional information

```python
def repro(self):
    expected = "hello"
    if sys.version_info < (3,):
        print(expected)
```

**Comment by karthiknadig** [comment]
Disabling sematic highlighting gets rid of the squiggle:
![image](https://user-images.githubusercontent.com/3840081/94068958-b74d9400-fda4-11ea-9b5f-8428b49f6b3a.png)

**Comment by huguesv** [comment]
I've implemented a fix, and it will be in the next release of Pylance.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.9.7, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202097-30-september-2020
